### PR TITLE
Added FIFO policy

### PR DESCRIPTION
**Pull request by shanthanu9**

### CODE REVIEW

**Comment by rohit-mp** [comment]
This works but I don't see the needs for few data structures used like queueOfBlocks and nextEmptyBlockInSet. 

Just maintaining a single value per set which can range from 0 to setAssociativity-1 would be sufficient to decide which block to replace.

Also, don't forget to remove the extra header files you've included.

**Comment by shanthanu9** [comment]
Single value per set? Consider what happens when a new block is inserted
(have to decide the next block to go out).

**Comment by rohit-mp** [comment]
The queue structure needed here can be implemented with an array and a single pointer. This queue (array) is the cache (set) itself and the pointer is the  only extra data structure needed here. Whatever the pointer points to is the block that gets replaced.

File: policies/fifo.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+void FIFO::update(ll block, int status){
+    if(status == 0) {
+        ll index = block/setAssociativity;
+        fifoBlockInSet[index] = (fifoBlockInSet[index]+1)%setAssociativity;
+    }
+}
+
+FIFO::~FIFO(){
```
Comment (rohit-mp) [inline policies/fifo.cpp]: Missing memory deallocation here

**Pull request merged**